src/main/java/modbus/SlaveFolder.java

```diff
@@ -814,31 +818,7 @@ protected JsonArray parseResponse(ReadResponse response, DataType dataType, doub
 		return retval;
 	}
 
-	static int toUnsignedInt(short x) {
-		return ((int) x) & 0xffff;
+	boolean checkConnectionByDevice() {
+		return true;

```

Comment:
yeah, could change it to ```return root.checkConnectionByDevice()```. Otherwise won't work properly when called from a subfolder.